src/ioabc.cpp

```diff
@@ -1021,6 +1025,94 @@ void ABCInput::InitScoreAndSection(Score *&score, Section *&section)
+                ++counter;
+                ++found;
+            }
+            --found;
+            sylType = sylLog_CON_u;
+        }
+        else if (abcLine.at(found) == '~') {
+            counter = 0;
+            sylType = sylLog_CON_s;
+        }
+        else if (abcLine.at(found) == '-') {
+            sylType = sylLog_CON_d;
+        }
+        else if (abcLine.at(found) == '\\') {
+            if ((found + 1 < abcLine.size()) && (abcLine.at(found + 1) == '-')) {
+                counter = 0;
+                ++found;
+                sylType = sylLog_CON_d;
+            }
+        }
+        // separate syllable from delimeters to form syl that we want to add
+        syllable = abcLine.substr(start, found - start);
+        syllable.erase(
+            std::remove_if(syllable.begin(), syllable.end(), [](unsigned char x) { return (x == '_') || (x == '\\'); }),
+            syllable.end());
+        if (!syllable.empty()) {
+            Text *sylText = new Text();
+            sylText->SetText(UTF8to16(syllable));
+            Syl *syl = new Syl();
+            syl->AddChild(sylText);
+            syl->SetCon(sylType);
+            if (sylType == sylLog_CON_d) {
+                syl->SetWordpos(sylLog_WORDPOS_m);
+            }
+            syllables.push_back({ syl, counter });
+        }
+
+        // find next delimeter in the string
+        start = found + 1;
+        found = abcLine.find_first_of(delimeters, start);
+        // if none found, the rest of the string is going to server as last syl
+        if ((found == std::string::npos) && (start < abcLine.size())) {
+            std::string syllable = abcLine.substr(start);
+            if (!syllable.empty() && syllable[syllable.size() - 1] == '\r') syllable.erase(syllable.size() - 1);
+            Text *sylText = new Text();
+            sylText->SetText(UTF8to16(syllable));
+            Syl *syl = new Syl();
+            syl->AddChild(sylText);
+            syl->SetCon(sylType);
+            syllables.push_back({ syl, counter });
+        }
+    }
+
+    // Iterate over notes and syllables simultaneously. Move through note array using counters for each syllable, moving
+    // for several notes if syllable needs to be held
+    for (size_t i = 0, j = 0; (i < m_lineNoteArray.size()) && (j < syllables.size()); ++j) {
+        while (m_lineNoteArray.at(i)->IsGraceNote() && (i < m_lineNoteArray.size())) {
+            ++i;
+        }
+        if (i >= m_lineNoteArray.size()) break;
+        Verse *verse = NULL;
+        verse = vrv_cast<Verse *>(m_lineNoteArray.at(i)->GetChild(0, VERSE));
+        if (!verse) {
+            verse = new Verse();
+            verse->SetN(m_verseNumber);
+            m_lineNoteArray.at(i)->AddChild(verse);
+        }
+        verse->AddChild(syllables.at(j).first);
+        i += syllables.at(j).second;
+    }
+    // increment verse number, in case next line in file is also w:
+    ++m_verseNumber;
```

Comment:
If not all the `Syl` are added to a `Note` then we have a memory leak. I guess you need to count how many are added, and delete the remaining ones by hand.